tests/core/test_link.py

```diff
@@ -0,0 +1,175 @@
+# Copyright (C) 2012 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+
+from conda.core import link
+from conda.models.records import PackageRecord
+
+
+def test_calculate_change_report_changed_variant():
+    """
+    Test to ensure that the change report will categorize a change in variant as
+    a "CHANGED" package.
+    """
+    unlink_precs = [
+        PackageRecord(
+            **{
+                "channel": "pkgs/main/linux-64",
+                "name": "mypackage",
+                "version": "2.3.9",
+                "build": "py35_0",
+                # notice the build number decrease between the unlink and link precs
```

Comment:
This would be a downgrade for me 🤔 Otherwise why are build number bumps reported as upgrades with the same version?